fastly/block_fastly_service_v1_s3logging.go

```diff
@@ -135,84 +159,103 @@ func processS3Logging(d *schema.ResourceData, conn *gofastly.Client, latestVersi
-			BucketName:                   sf["bucket_name"].(string),
-			AccessKey:                    sf["s3_access_key"].(string),
-			SecretKey:                    sf["s3_secret_key"].(string),
-			Period:                       uint(sf["period"].(int)),
-			GzipLevel:                    uint(sf["gzip_level"].(int)),
-			Domain:                       sf["domain"].(string),
-			Path:                         sf["path"].(string),
-			Format:                       sf["format"].(string),
-			FormatVersion:                uint(sf["format_version"].(int)),
-			TimestampFormat:              sf["timestamp_format"].(string),
-			ResponseCondition:            sf["response_condition"].(string),
-			MessageType:                  sf["message_type"].(string),
-			Placement:                    sf["placement"].(string),
-			ServerSideEncryptionKMSKeyID: sf["server_side_encryption_kms_key_id"].(string),
+		log.Printf("[DEBUG] Create S3 Logging Opts: %#v", opts)
+		_, err := conn.CreateS3(&opts)
+		if err != nil {
+			return err
 		}
+	}
+	return nil
+}
+
+func readS3Logging(conn *gofastly.Client, d *schema.ResourceData, s *gofastly.ServiceDetail) error {
+	log.Printf("[DEBUG] Refreshing S3 Logging for (%s)", d.Id())
+	s3List, err := conn.ListS3s(&gofastly.ListS3sInput{
+		Service: d.Id(),
+		Version: s.ActiveVersion.Number,
+	})
+
+	if err != nil {
+		return fmt.Errorf("[ERR] Error looking up S3 Logging for (%s), version (%v): %s", d.Id(), s.ActiveVersion.Number, err)
+	}
 
-		redundancy := strings.ToLower(sf["redundancy"].(string))
-		switch redundancy {
-		case "standard":
-			opts.Redundancy = gofastly.S3RedundancyStandard
-		case "reduced_redundancy":
-			opts.Redundancy = gofastly.S3RedundancyReduced
+	sl := flattenS3s(s3List)
+
+	if err := d.Set("s3logging", sl); err != nil {
+		log.Printf("[WARN] Error setting S3 Logging for (%s): %s", d.Id(), err)
+	}
+	return nil
+}
+
+func flattenS3s(s3List []*gofastly.S3) []map[string]interface{} {
+	var sl []map[string]interface{}
+	for _, s := range s3List {
+		// Convert S3s to a map for saving to state.
+		ns := map[string]interface{}{
+			"name":                              s.Name,
+			"bucket_name":                       s.BucketName,
+			"s3_access_key":                     s.AccessKey,
+			"s3_secret_key":                     s.SecretKey,
+			"path":                              s.Path,
+			"period":                            s.Period,
+			"domain":                            s.Domain,
+			"gzip_level":                        s.GzipLevel,
+			"format":                            s.Format,
+			"format_version":                    s.FormatVersion,
+			"timestamp_format":                  s.TimestampFormat,
+			"redundancy":                        s.Redundancy,
+			"response_condition":                s.ResponseCondition,
+			"message_type":                      s.MessageType,
+			"placement":                         s.Placement,
+			"server_side_encryption":            s.ServerSideEncryption,
```

Comment:
These values are API response values, hence why we don't have to do the `switch` statement.